File: JavaScript/fibonacciNumber.js
/**
 * @param {number} N
 * @return {number}
 */
 // Recursive solution
var fib = function(N) {

    if (N <= 1) {
        return N;
    }

    return fib(N-1) + fib(N-2);
};

// Iterative solution
var fib = function(N) {

    if (N <= 1) {
        return N;
    }

    let fib = 1;
    let prevFib = 1;

    for (i = 2; i < N; i++) {
        let temp = fib;
        fib = prevFib + fib;
        prevFib = temp;
    }
    return fib;
};

File: Ruby/fibonacciNumber.rb
# @param {Integer} n
# @return {Integer}
def fib(n)

    if n <= 1
        return n
    end

    return fib(n-1) + fib(n-2)
end

# Iterative solution

def fib(n)

    if n <= 1
        return n
    end

    fib = 1
    prevFib = 1

    i = 2

    while i < n
        temp = fib
        fib = prevFib + fib
        prevFib = temp
        i+=1
    end

    return fib

end
